File: Reversing the equation - GFG/reversing-the-equation.py
#User function Template for python3

class Solution:
    def reverseEqn(self, s):
        # code here
        x = ['*','-','/','+']
        l = []
        r = ""
        ans = ""
        
        for i in s:
            if i not in x:
                r = r + i
            else:
                l.append(r)
                r = ""
                l.append(i)
                
        l.append(r) 
        #print(l)# for the characters that were present after the last operator
        l.reverse()
        #print(l)
        for i in range(len(l)):
            ans += l[i]
        
        return ans

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        s = input().strip()
        ob = Solution()
        ans = ob.reverseEqn(s)
        print(ans)
# } Driver Code Ends

File: Reversing the equation - GFG/reversing-the-equation.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out=new PrintWriter(System.out);
        int t = Integer.parseInt(in.readLine().trim());
        while(t-- > 0)
        {
            String s;
            s = in.readLine().trim();
            
            Solution ob = new Solution();
            
            out.println(ob.reverseEqn(s));    
        }
        out.close();
    }
}
// } Driver Code Ends


//User function Template for Java
class Solution
{
   
    String reverseEqn(String S)
    {
        // your code here
        StringBuilder sb = new StringBuilder();
        StringBuilder ans = new StringBuilder();
        Stack<String> s = new Stack<>();
        for(int i=0;i<S.length();i++){
            char ch =S.charAt(i);
            if(ch == '+' || ch == '-' || ch == '*' || ch == '/'){
                s.push(ans.toString());
                s.push(Character.toString(ch));
                ans.setLength(0);
            }else{
                ans.append(ch);
            }
        }
        s.push(ans.toString());
        while(!s.isEmpty()){
            sb.append(s.pop());
        }
        return sb.toString();
    }
}